### PR TITLE
simplify isSentToMultisig

### PR DESCRIPTION
**Pull request by matthewleon**
Successor of https://github.com/bitcoinj/bitcoinj/pull/1864

### CODE REVIEW

**Comment by matthewleon** [comment]
Actually still not 100% sure this is documented and working correctly. Different sources online switch the `m` and the `n`. I'll have to compare to the core source.

**Comment by matthewleon** [comment]
Have amended the javadoc. I believe the `m` and the `n` were switched in the doc, but functioning correctly in the code. Please check my thinking here carefully. Thanks!

**Comment by schildbach** [comment]
I agree the JavaDoc has it the wrong way round. See https://github.com/bitcoin/bips/blob/master/bip-0011.mediawiki

File: core/src/main/java/org/bitcoinj/script/ScriptPattern.java [hunk 1]
```diff
@@ -198,7 +198,7 @@ public static boolean isP2WSH(Script script) {
 
     /**
      * Returns whether this script matches the format used for multisig outputs:
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/script/ScriptPattern.java]: Perhaps we should use the tern "M-of-N multisig" here? This would define what m and n actually means.

And the Javadoc change could go on its own commit, as it's not related to the exception change as far as I understand.

Comment (matthewleon) [reply core/src/main/java/org/bitcoinj/script/ScriptPattern.java]: https://github.com/bitcoinj/bitcoinj/pull/1867

**Comment by matthewleon** [comment]
It might also make sense to merge https://github.com/bitcoinj/bitcoinj/pull/1863 first.

**Comment by schildbach** [comment]
Thanks a lot! I just merged this.
